change GetFleetSpeedForMission for ParkInThatAlly MissionID

0xE232FE opened the pull request:
Add case for speedFleetHolding MissionID to get correct FlightTime.

alaingilbert commented:
Not sure why you added the `FuelCapacity` to all ships which has the same value as`BaseCargoCapacity` ?

0xE232FE commented:
Oh yes I see what you mean in EspionageProbe.go you set the baseCargoCapacity to 5 which I can use for that. I completely oversaw this! I simplified this to just use the BaseCargoCapacity in the GetFuelCapacity function! 


```
// GetFuelCapacity returns ship fuel capacity
func (b BaseShip) GetFuelCapacity() int64 {
	return b.BaseCargoCapacity
}
```
At first I wanted to give a preview of the FuelConsumption vs FuelCapacity like the OGame Website. Deuterium Consumption turning red if the capacity is over 100%. So we could prevent the Send-Fleet Function to send fleet if there is not enough fuel Capacity. I know that the Endpoint tells us there is not enough cargo, but I would just to prevent this from sending to OGame Enpoint.

![image](https://user-images.githubusercontent.com/21330562/131630438-9c84e2b7-07d0-472e-ba17-bfe960329a6b.png)

I also added an Extractor function for Fleet1 Page to get the Researches needed for the Fleet calculations before sending the fleet and we can give better error that we need more cargo because the FuelCapacity is not enough.

alaingilbert commented on baseShip.go:
```diff
@@ -41,6 +41,11 @@ func (b BaseShip) GetFuelConsumption(techs Researches, fleetDeutSaveFactor float
 }
 
+// GetFuelCapacity returns ship fuel capacity
+func (b BaseShip) GetFuelCapacity() int64 {
```
Unless ogame changed the mechanics, the amount of fuel that you can carry should increase according to your techs and character class.
Basically, the function is already implemented (`Cargo` returns you the total cargo available) -> https://github.com/alaingilbert/ogame/blob/b2ba1299e6ce5001c80ff11d643c03bf28cf6e8d/ships.go#L119-L124

0xE232FE replied:
I checked the HTML JavaScript of their Calculation and they use var ShipsData which has different Values for baseFuelCapacity and baseCargoCapacity (the CargoCapacity already have the HyperSpace Bonus added so they display the correct cargo on the Page) and in the FuelCapacity computation they really use the baseFuelCapacity which is not equal the baseCargoCapacity in the HTML ShipsData variable. Check it on fleetDispatch page there is the JSON shipsData variable which contains all computed values for the Cargos. Thats the point where I know that the FuelCapacity is not increased by the Tech Hyperspace Technology.

```
  "204": {
    "id": 204,
    "name": "Light Fighter",
    "baseFuelConsumption": 20,
    "baseFuelCapacity": 50,
    "baseCargoCapacity": 62,
    "fuelConsumption": 20,
    "baseSpeed": 12500,
    "speed": 22500,
    "cargoCapacity": 62,
    "fuelCapacity": 50,
    "number": 1,
    "recycleMode": 0
  }
```

For example 1x LightFighter consumes 138 Fuel but its fuel Capacity is 50. Which leads into
138 / 50 * 100 = 276% (not equal) 138 / 62 * 100 = 222%

And this means it will not be increased by the HyperSpaceTechnology

![image](https://user-images.githubusercontent.com/21330562/131663044-ba5aed1c-0f2b-4c9e-979b-8d7cf0c3750b.png)

0xE232FE replied:
I tried to send 1x LightFigher to a mission FuelConsumption 52 and CargoCapacity 62  so 52 is between 50 and 62. The mission fails with not enough cargo space available. So they use 50 for the FuelCapacity of the LightFighter internally. I remember as they added the Hyper Space Technology to increase Cargo Space the disconnected the Cargo from the FuelCapacity. And they only increase the Cargo and not the Fuel Capacity, which means the ships can have a higher range which was not wanted.

0xE232FE replied:
Now I got you!
```
fuelCapacity := ogame.ShipsInfos{EspionageProbe: 1}.Cargo(ogame.Researches{}, true, false, false)
// fuelCapacity = 5
```

0xE232FE commented:
Now I got you!

```
fuelCapacity := ogame.ShipsInfos{EspionageProbe: 1}.Cargo(ogame.Researches{}, true, false, false)
// fuelCapacity = 5
```

0xE232FE commented on ships.go:
```diff
@@ -69,7 +69,7 @@ func (s ShipsInfos) HasFlyableShips() bool {
 	var minSpeed int64 = math.MaxInt64
 	for _, ship := range Ships {
-		if ship.GetID() == SolarSatelliteID {
+		if ship.GetID() == SolarSatelliteID || ship.GetID() == CrawlerID {
```
Added CrawlerID it is another not movable Ship Type

0xE232FE commented on ogame.go:
```diff
@@ -4218,6 +4222,18 @@ func (b *OGame) sendFleet(celestialID CelestialID, ships []Quantifiable, speed S
+	_, fuel := CalcFlightTime(
+		b.GetCachedCelestialByID(celestialID).GetCoordinate(), where,
+		b.serverData.Galaxies, b.serverData.Systems, b.serverData.DonutGalaxy, b.serverData.DonutSystem, b.serverData.GlobalDeuteriumSaveFactor,
+		float64(speed)/10, GetFleetSpeedForMission(b.IsV81(), b.serverData, mission), ShipsInfos{}.FromQuantifiables(ships), techs, b.characterClass, holdingTime)
```
In CalcFlightTime() the speed must be divided by 10 `float64(speed/10)` this should fix the calculation finally.

0xE232FE commented on handlers.go:
```diff
@@ -1029,6 +1033,8 @@ func GetFromGameHandler(c echo.Context) error {
 	}
 	pageHTML, _ := bot.GetPageContent(vals)
 	pageHTML = replaceHostname(bot, pageHTML)
+	pageHTML = removeCookieBanners(pageHTML)
```
Remove the cookie Banner from HTML.
![image](https://user-images.githubusercontent.com/21330562/131993536-1fede11c-eb41-45d3-bc9c-320cdef07695.png)

0xE232FE replied:
Remove Gameforge Cookie Banner from HTML
![image](https://user-images.githubusercontent.com/21330562/131993780-ab6a82d4-7da9-4275-bd1a-1e597800c106.png)
